libs/execution/include/hpx/execution/executors/static_chunk_size.hpp

```diff
@@ -58,16 +58,28 @@ namespace hpx { namespace parallel { namespace execution {
         {
             // use the given chunk size if given
             if (chunk_size_ != 0)
+            {
                 return chunk_size_;
+            }
+
+            if (cores == 1)
+            {
+                return 1;
+            }
 
             // Make sure the internal round robin counter of the executor is
             // reset
             execution::reset_thread_distribution(*this, exec);
 
-            // by default use static work distribution over number of
-            // available compute resources, create four times the number of
-            // chunks than we have cores
-            return (num_tasks + 4 * cores - 1) / (4 * cores);    // -V112
+            // Return a chunk size that is a power of two; and that leads to at
+            // least 2 chunks per core, and at most 4 chunks per core.
```

Comment:
I haven't compared e.g. chunk sizes that lead to 4-8 (or even more) chunks per core. I thought 2-4 per core would be a conservative number for good performance even though in the worst case 2 might lead to considerable load imbalance.

Reply:
@scheherzade can you comment on this?

Reply:
I should note that the idea for powers of two comes from Kokkos. Kokkos gets by with much higher number of chunks per core (up to 100), because they don't end up with a task per chunk.

We could make this 4-8 which would mean *at least* as many chunks per core as we have on master right now.